lib/src/linting/handle-lint-violations.spec.ts

```diff
@@ -0,0 +1,220 @@
+import { handleLintViolations } from "./handle-lint-violations";
+
+describe("handle lint violations", () => {
+  let deferExit: boolean;
+  let errorMock: jest.Mock;
+  let warnMock: jest.Mock;
+
+  describe("lint violation when rule not specified", () => {
+    beforeEach(() => {
+      errorMock = jest.fn();
```

Comment:
This is done in each `describe()` block, I think it would be better to extract it out into a top-level `beforeEach()`?